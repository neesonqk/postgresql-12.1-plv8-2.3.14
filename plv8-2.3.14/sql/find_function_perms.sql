-- test find_function permissions failure
CREATE FUNCTION perm() RETURNS void AS $$ plv8.elog(NOTICE, 'nope'); $$ LANGUAGE plv8;

CREATE ROLE someone_else;

REVOKE EXECUTE ON FUNCTION perm() FROM public;

SET ROLE TO someone_else;
DO $$ const func = plv8.find_function('perm') $$ LANGUAGE plv8;
DO $$ const func = plv8.find_function('perm()') $$ LANGUAGE plv8;

RESET ROLE;
DROP ROLE someone_else;

DROP FUNCTION perm();
