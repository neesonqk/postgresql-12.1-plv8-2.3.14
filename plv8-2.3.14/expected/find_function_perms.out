-- test find_function permissions failure
CREATE FUNCTION perm() RETURNS void AS $$ plv8.elog(NOTICE, 'nope'); $$ LANGUAGE plv8;
CREATE ROLE someone_else;
REVOKE EXECUTE ON FUNCTION perm() FROM public;
SET ROLE TO someone_else;
DO $$ const func = plv8.find_function('perm') $$ LANGUAGE plv8;
WARNING:  failed to find or no permission for js function perm
DO $$ const func = plv8.find_function('perm()') $$ LANGUAGE plv8;
WARNING:  failed to find or no permission for js function perm()
RESET ROLE;
DROP ROLE someone_else;
DROP FUNCTION perm();
